client/src/pages/TaskHistoryPage/TaskHistoryPage.js

```diff
@@ -1,23 +1,71 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import styles from "./TaskHistoryPage.module.css";
 import TaskListView from "../../components/TaskListView/TaskListView.js";
+import {
+  buildDailyTaskRoute,
+  buildDateRangeRoute,
+  convertTaskData,
+  getWeekRange,
+  getMonthRange,
+} from "../../utils.js";
+import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage.js";
+import { instance } from "../../axios.js";
 
 const TaskHistoryPage = () => {
-  const sampleTask = {
-    title: "gdsc meeting",
-    location: "dh2020",
-    startTime: "05:00pm",
-    endTime: "07:00pm",
-    isOngoing: false,
+  const dayFilter = "day";
+  const weekFilter = "week";
+  const monthFilter = "month";
+  const allFilter = "all";
+  const [filter, setFilter] = useState(dayFilter);
+  const [taskData, setTaskData] = useState([]);
+  const [loadingError, setLoadingError] = useState(false);
+  const [loadingErrorMessage, setLoadingErrorMessage] = useState("");
+
+  const getTaskOnRoute = async (route) => {
+    setLoadingErrorMessage("");
+    await instance
+      .get(route)
+      .then((taskData) => {
+        console.log(taskData);
+        setTaskData();
+        setLoadingError(false);
+      })
+      .catch(() => {
+        setLoadingErrorMessage("Unable to load tasks!");
+        setLoadingError(true);
+      });
   };
-  const sampleTasks = [];
-  for (let i = 0; i < 15; i++) {
-    sampleTasks.push({ ...sampleTask });
-  }
+
+  useEffect(() => {
+    let startDate, endDate;
+    switch (filter) {
+      case dayFilter:
+        getTaskOnRoute(buildDailyTaskRoute(new Date()));
+        break;
+      case weekFilter:
+        [startDate, endDate] = getWeekRange();
+        getTaskOnRoute(buildDateRangeRoute(startDate, endDate));
+        break;
+      case monthFilter:
+        [startDate, endDate] = getMonthRange();
+        getTaskOnRoute(buildDateRangeRoute(startDate, endDate));
+        break;
+      case allFilter:
```

Comment:
I think for `allFilter`, it can be left as a default case

Reply:
Almost acting like a "catch all otherwise"

Reply:
Changed to default case.